Sming/Libraries/OtaUpgrade/BasicOtaUpgradeStream.cpp

```diff
@@ -0,0 +1,251 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2017 by Slavey Karadzhov
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * BasicOtaUpgradeStream.cpp
+ *
+ ****/
+
+#include "OtaUpgradeStream.h"
+#include <algorithm>
+#include <esp_spi_flash.h>
+#include <Data/HexString.h>
+#include <FlashString/Array.hpp>
+
+#ifndef ENABLE_OTA_DOWNGRADE
+extern "C" const uint64_t OTA_BuildTimestamp PROGMEM;
```

Comment:
The definition is in a generated file, which is currently a .c file, because it does not require any C++ features, but I can also change it into a .cpp file.

Reply:
NB. The `PROGMEM` is redundant here.